internal/provisioner/objects/service/service.go

```diff
@@ -303,17 +287,20 @@ func DesiredEnvoyService(contour *model.Contour) *corev1.Service {
 		}
 	case model.NodePortServicePublishingType:
 		svc.Spec.Type = corev1.ServiceTypeNodePort
-		if len(contour.Spec.NetworkPublishing.Envoy.NodePorts) > 0 {
-			for _, p := range contour.Spec.NetworkPublishing.Envoy.NodePorts {
-				if p.PortNumber != nil {
-					for i, q := range svc.Spec.Ports {
-						if q.Name == p.Name {
-							svc.Spec.Ports[i].NodePort = *p.PortNumber
-						}
-					}
+		if len(contour.Spec.NetworkPublishing.Envoy.NodePorts) == 0 {
+			break
+		}
```

Comment:
no need to explicitly check for an empty slice here, if `NodePorts` is nil/empty the subsequent loop will be a no-op anyway.